### PR TITLE
Add optionsfactory

### PR DESCRIPTION
**Pull request by johnomotani**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by johnomotani** [comment]
@conda-forge-admin, please ping team
My first conda-forge package - I think it's ready for merge now. Thanks!

File: recipes/optionsfactory/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+
+test:
+  requires:
```
Comment (hmaarrfk) [inline recipes/optionsfactory/meta.yaml:25]:
```suggestion
  imports:
    - optionsfactory
  requires:
```

File: recipes/optionsfactory/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
```
Comment (hmaarrfk) [inline recipes/optionsfactory/meta.yaml:20]:
```suggestion
```

Comment (hmaarrfk) [reply recipes/optionsfactory/meta.yaml:20]: does your package actually need pip?

Comment (johnomotani) [reply recipes/optionsfactory/meta.yaml]: No, it doesn't actually need pip. Sorry, that line was just copied from the example!

Comment (xhochy) [reply recipes/optionsfactory/meta.yaml]: It does. `pip` is used for building this package.

Comment (johnomotani) [reply recipes/optionsfactory/meta.yaml]: :+1: Thanks @xhochy

File: recipes/optionsfactory/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: 43f0c4ace9ecde6e755aaeef0bcd9ff78c2503e85d189c6db51c7b8387183819
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.6
```
Comment (johnomotani) [inline recipes/optionsfactory/meta.yaml:19]:
```suggestion
    - python >=3.6
    - pip
```

**Pull request merged**